src/components/templates/ScreenShare/ScreenShare.tsx

```diff
@@ -0,0 +1,151 @@
+  channel: "videotest",
+  token: "",
+};
+
+export const ScreenShare = () => {
+  const [openLeaveStageModal, setOpenLeaveStageModal] = useState(false);
+  const remoteUsers = useAgoraRemotes(remotesClient, AGORA_CHANNEL);
+
+  const {
+    localCameraTrack,
+    toggleCamera,
+    toggleMicrophone,
+    isCameraOn,
+    isMicrophoneOn,
+    closeTracks,
+    onJoin: cameraClientJoin,
+  } = useAgoraCamera(cameraClient, AGORA_CHANNEL);
+
+  const { localScreenTrack, shareScreen, stopShare } = useAgoraScreenShare(
+    screenClient,
+    AGORA_CHANNEL
+  );
+
+  return (
+    <div className="ScreenShare">
+      <div className="ScreenShare__scene">
+        <div className="ScreenShare__scene--players">
+          {localCameraTrack && (
+            <div>
+              <Player
+                videoTrack={localCameraTrack}
+                showButtons
+                isCamOn={isCameraOn}
+                isMicOn={isMicrophoneOn}
+                isSharing={!!localScreenTrack}
+                toggleCam={toggleCamera}
+                toggleMic={toggleMicrophone}
+              />
+            </div>
+          )}
+          {localScreenTrack && (
+            <div>
+              <Player videoTrack={localScreenTrack} />
+            </div>
+          )}
+          {remoteUsers.map(
+            (user) =>
+              user.uid !== screenClient.uid &&
+              user.uid !== cameraClient.uid && (
+                <div key={user.uid}>
+                  {user.hasVideo && (
+                    <Player
+                      videoTrack={user.videoTrack}
+                      audioTrack={user.audioTrack}
+                    />
+                  )}
+                </div>
+              )
+          )}
+        </div>
+
+        <div className="ScreenShare__scene--buttons">
+          {cameraClient.connectionState === "CONNECTED" ? (
+            <>
+              <Button
+                onClick={() => {
+                  localScreenTrack ? stopShare() : shareScreen();
+                }}
+                rightLabel={localScreenTrack && "You are screensharing"}
+                variant="secondary"
+                small
+              >
+                {localScreenTrack ? "Stop Sharing" : "Share Screen"}
+              </Button>
+
+              <Button
+                onClick={() => {
+                  setOpenLeaveStageModal(true);
+                }}
+                leftLabel="You are on stage"
+                variant="secondary"
+                small
+              >
+                Leave Stage
+              </Button>
+            </>
+          ) : (
+            <div className="ScreenShare__scene--buttons--join-stage">
+              <Button
+                onClick={() => {
+                  cameraClientJoin();
+                  screenClient.join(
+                    AGORA_CHANNEL.appId,
+                    AGORA_CHANNEL.channel,
+                    AGORA_CHANNEL.token
+                  );
+                }}
```

Comment:
useCallback

Reply:
This seems to have a bunch of 'implementation details' looking things that should probably be wrapped up/abstracted within the hooks more too. So that might be the place to make the `useCallback` or similar?